### DISCUSSION TITLE
GitHub Workflows for CODEOWNERS for Spices

### DISCUSSION THREAD

**Discussion by rcalixte** [linuxmint/discussions#65] [Ideas]
Currently, `CODEOWNERS` files require manual updating and are frequently out-of-date.

Latest update pull requests:
linuxmint/cinnamon-spices-applets#4925
linuxmint/cinnamon-spices-desklets#859
linuxmint/cinnamon-spices-extensions#468
linuxmint/cinnamon-spices-themes#753

We can instead create a workflow that will auto-run the `codeowners_creator.py` script after a commit is merged to master and auto-open pull requests if there is a diff on the `CODEOWNERS` file. (No action would be taken if there was no change to the CODEOWNERS file.)

**Comment by rcalixte** [comment]
@schorschii I'm thinking we can include the notification changes here. There are options to improve all elements of the current system. Let's take a diligent approach here.

**Reply by schorschii** [reply]
OK, it's nice to automatically update the `CODEOWNERS` file. But how does this solve the problem that issue reporters and pull request authors do not look in that file and simply notify the user named in the desklet UUID?

In the specific case, I was already mentioned as maintainer in the `CODEOWNERS` file, but both [issue](https://github.com/linuxmint/cinnamon-spices-desklets/issues/769) and [pull request](https://github.com/linuxmint/cinnamon-spices-desklets/pull/921) didn't notified me in the first place.

One additional sentence which explicitly points this out in the templates is necessary in my opinion, as suggested in [my PR](https://github.com/linuxmint/cinnamon-spices-desklets/pull/923).

**Reply by rcalixte** [reply]
I'm suggesting more structure than human actions. I think we can leverage it for notifications too. Given that the issue was over a year old as well, I'm contemplating if this should even be applied retroactively.

**Reply by schorschii** [reply]
If notifying the correct maintainer is possible automatically, that would surely be nice. But this is certainly not an easy thing which is implemented tomorrow. Why not add a simple sentence of clarification to the issue/pr templates until the time comes?

**Reply by rcalixte** [reply]
It will be overlooked just the same. (The same is already in the existing documentation and it isn't always observed.) And adding something with the sole purpose of reverting it feels wasteful. I have other obligations but I can certainly try to rush this to tomorrow if that would suffice. 😄

**Reply by rcalixte** [reply]
I do agree that we can adjust the text portions as well. Let's approach this holistically though.

**Reply by schorschii** [reply]
I don't think it would be the same. The issue/pr template is always seen by the issue/pr author. Only a minority will read documentation before opening issues/PRs.

Don't rush, please take the necessary time. This is why I suggest a simple adjustment of the template text first.

**Reply by schorschii** [reply]
When do you plan to update the issue/pr templates @rcalixte ?

**Reply by rcalixte** [reply]
PRs opened here:
linuxmint/cinnamon-spices-applets#5038
linuxmint/cinnamon-spices-desklets#925
linuxmint/cinnamon-spices-extensions#486
linuxmint/cinnamon-spices-themes#759

This is a first pass at the Issue form/template. Once we get something ready to be merged, we can also create an equivalent for Pull Requests.

I need to spend time to get the automation implemented as the team desires but after that, we should be in the clear here.
